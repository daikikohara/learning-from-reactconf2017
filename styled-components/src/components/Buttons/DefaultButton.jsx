import styled from 'styled-components';

const Button = styled.button`
  background: ${props => props.active ? 'gray' : 'white'};
  color: ${props => props.active ? 'white' : 'black'};
  margin: 1rem;
  font-size: 2rem;
  padding: 0.25rem 1rem;
  border: 2px solid ${props => props.active ? 'transparent' : 'black'};
  border-radius: 3px;
  &:hover {
    cursor: pointer;
  }
`;

export default Button;

