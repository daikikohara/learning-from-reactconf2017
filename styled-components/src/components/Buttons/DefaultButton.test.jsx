import React from 'react';
import {matcher, serializer} from 'jest-styled-components';
import renderer from 'react-test-renderer';
import DefaultButton from './DefaultButton';

expect.extend(matcher);
expect.addSnapshotSerializer(serializer);

describe('DefaultButton Test', () => {
  it('should be rendered correctly on active', () => {
    const renderedElement = renderer.create(<DefaultButton active={true} />);
    expect(renderedElement).toMatchStyledComponentsSnapshot();
  });
  it('should be rendered correctly on disabled', () => {
    const renderedElement = renderer.create(<DefaultButton active={false} />);
    expect(renderedElement).toMatchStyledComponentsSnapshot();
  });
});

