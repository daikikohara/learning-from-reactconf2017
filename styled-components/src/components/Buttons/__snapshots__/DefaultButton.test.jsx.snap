// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultButton Test should be rendered correctly on active 1`] = `
.jXtbih {
  background: gray;
  color: white;
  margin: 1rem;
  font-size: 2rem;
  padding: 0.25rem 1rem;
  border: 2px solid transparent;
  border-radius: 3px;
}

.jXtbih:hover {
  cursor: pointer;
}

<button
  className="jXtbih"
/>
`;

exports[`DefaultButton Test should be rendered correctly on disabled 1`] = `
.auoqs {
  background: white;
  color: black;
  margin: 1rem;
  font-size: 2rem;
  padding: 0.25rem 1rem;
  border: 2px solid black;
  border-radius: 3px;
}

.auoqs:hover {
  cursor: pointer;
}

<button
  className="auoqs"
/>
`;
